browser/misc_metrics/page_metrics_tab_helper.cc

```diff
@@ -26,13 +27,18 @@ PageMetricsTabHelper::PageMetricsTabHelper(content::WebContents* web_contents)
   page_metrics_ = ProfileMiscMetricsServiceFactory::GetServiceForContext(
                       web_contents->GetBrowserContext())
                       ->GetPageMetrics();
-  DCHECK(page_metrics_);
+  if (!page_metrics_) {
+    CHECK_IS_TEST();
+  }
 }
 
 PageMetricsTabHelper::~PageMetricsTabHelper() = default;
 
 void PageMetricsTabHelper::DidFinishNavigation(
     content::NavigationHandle* navigation_handle) {
+  if (!page_metrics_) {
```

Comment:
this is a bit fragile because page_metrics_ could be used in other places, I think it would be better to have `MaybeCreateForWebContents` and do nothing if it's null

Reply:
I guess it already has this check in `PageMetricsTabHelper::CheckNavigationEvent` so maybe this is fine